pynicotine/shares.py

```diff
@@ -95,7 +95,8 @@ class DatabaseError(Exception):
 class Database:
     """Custom key-value database format for Nicotine+ shares."""
 
-    FILE_HEADER = b"N+DB"
+    FILE_SIGNATURE = b"N+DB"
+    VERSION = 1
```

Comment:
Why not just align this value with the program version, that is from `config.version`?

It is acceptable for a full database rebuild to occur whenever the program version is upgraded, regardless if there will be any changes to the database structure or not.

Reply:
What's the point of forcing a rebuild if there haven't been any changes made to the format? Seems wasteful to do it for every minor release of Nicotine+, especially if we make those more frequent again.

Reply:
Fair enough. Why is this not version 3?

Reply:
I suppose it's debatable whether we should reset the version or not. Since it's a completely new format unrelated to and incompatible with the old one, it seemed appropriate to reset the version.

Reply:
Another thing that came to mind is if we should use a new file extension for these databases, and remove the ones from previous Nicotine+ versions (or should we leave them?).

Reply:
> maybe .ndb

That is already used by some known application see https://fileinfo.com/extension/ndb 

Suggest .dbn is more associated with binary data files, yet it isn't so common.

> should we leave them?

It seems prudent to keep the user's machines as tidy as possible. There is no real value in keeping obsolete data lying around, potentially forever.

Reply:
> Well then let's set config.version and GitHub releases back to Nicotine+ `1.0.0` !

I've reverted the version to 3, since we won't necessarily make changes to the db structure when bumping the version in the future (could happen if we make changes to the file metadata, for example).

> Suggest .dbn is more associated with binary data files, yet it isn't so common.

Done.

> It seems prudent to keep the user's machines as tidy as possible. There is no real value in keeping obsolete data lying around, potentially forever.

And deleted.